sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/PostLedgerEntryOperation.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Security.ConfidentialLedger
+{
+    /// <summary>
+    /// Tracks the status of a call to <see cref="ConfidentialLedgerClient.PostLedgerEntry(Azure.Core.RequestContent,string,bool,Azure.RequestOptions)"/> and <see cref="ConfidentialLedgerClient.PostLedgerEntryAsync(Azure.Core.RequestContent,string,bool,Azure.RequestOptions)"/>
+    /// until completion.
+    /// </summary>
+    public class PostLedgerEntryOperation : Operation
+    {
+        private readonly ConfidentialLedgerClient _client;
+        private Response _response;
+        private bool _hasCompleted;
+        private OperationFailedException _exception;
+
+        /// <summary>
+        /// Initializes a previously run operation with the given <paramref name="transactionId"/>.
+        /// </summary>
+        /// <param name="client"> Tje <see cref="ConfidentialLedgerClient"/>. </param>
+        /// <param name="transactionId"> The transaction id from a previous call to
+        /// <see cref="ConfidentialLedgerClient.PostLedgerEntry(Azure.Core.RequestContent,string,bool,Azure.RequestOptions)"/>. </param>
+        public PostLedgerEntryOperation(ConfidentialLedgerClient client, string transactionId)
+        {
+            _client = client;
+            Id = transactionId;
+        }
+
+        /// <summary>
+        /// A constructor for mocking.
+        /// </summary>
+        protected PostLedgerEntryOperation()
+        { }
+
+        /// <inheritdoc />
+        public override Response GetRawResponse() => _response;
+
+        /// <inheritdoc />
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+        {
```

Comment:
Use the OperationInternal for the implementation. (https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/OperationInternal.cs)

It will do all the right things for you.

Reply:
cc @kinelski

Reply:
cool - Does this make sense? https://github.com/Azure/azure-sdk-for-net/issues/22426

Reply:
Good idea.